tools/eslint-rules/only-ascii-characters.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Prohibit the use of non-ascii characters
+ * @author Kalon Hinds
+ */
+
+/* eslint no-control-regex:0 */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const nonAsciiPattern = new RegExp('([^\x00-\x7F])', 'g');
+const suggestions = {
+  '’': '\'',
+  '—': '-'
+};
+
+module.exports = {
+  create: function(context) {
+    return {
+      Program: (node) => {
+        const errors = [];
+        const source = context.getSourceCode();
+        const sourceTokens = source.getTokens(node);
+        const commentTokens = source.getAllComments();
+        const tokens = sourceTokens.concat(commentTokens);
+
+        tokens.forEach((token) => {
+          const { value } = token;
+          const matches = value.match(nonAsciiPattern);
+
+          if (matches) {
+            const { loc } = token;
+
+            // Will only report the first non-ascii character per line
+            const character = matches[0];
```

Comment:
This [line](https://github.com/hkal/node/blob/e7b2a191adecc4efe32f7509d31b327d684a65d3/tools/eslint-rules/only-ascii-characters.js#L44) makes the values in the `tokens` (couldn't think of a better name) array look like:

```javascript
{ type: 'Punctuator',
  value: ';',
  start: 22327,
  end: 22328,
  loc:
   SourceLocation {
     start: Position { line: 767, column: 1 },
     end: Position { line: 767, column: 2 } },
  range: [ 22327, 22328 ] }
{ type: 'Line',
  value: ' Copyright Joyent, Inc. and other Node contributors.',
  start: 0,
  end: 54,
  range: [ 0, 54 ],
  loc:
   { start: Position { line: 1, column: 0 },
     end: Position { line: 1, column: 54 } } }
```

In the case of type `Line` we only report the the first occurrence of a non-ASCII character. In my latest iteration I took the comment out since it didn't really help clarify.